doc/source/user/basics.creation.rst

```diff
@@ -9,138 +9,353 @@ Array creation
-docstring for complete information on the various ways it can be used. A few
-examples will be given here: ::
+::
 
- >>> np.arange(10)
+ >>> a = array([127,128,129], dtype = int8)
+ >>> print(a)
+ [ 127 -128 -127 ]
+
+An 8-bit signed integer represents integers from -128 to 127.
+Assigning the ``int8`` array to integers outside of this range results
+in overflow. This feature can often be misunderstood. If you
+perform calculations with mismatching ``dtypes``, you can get unwanted
+result,  for example: ::
+
+    >>> a = array([2, 3, 4], dtype = uint32)
+    >>> b = array([5, 6, 7], dtype = uint32)
+    >>> c_unsigned32 = a - b
+    >>> print('unsigned c:', c_unsigned32, c_unsigned32.dtype)
+    unsigned c: [4294967293 4294967293 4294967293] uint32
+    >>> c_signed32 = a - b.astype(int32)
+    >>> print('signed c:', c_signed32, c_signed32.dtype)
+    signed c: [-3 -3 -3] int64
+
+Notice when you perform operations with two arrays of the same
+``dtype``: ``uint32``, the resulting array is the same type. When you
+perform operations with different ``dtype``, NumPy will 
+assign a new type that satisfies all of the array elements involved in
+the computation, here ``uint32`` and ``int32`` can both be represented in
+as ``int64``. 
+
+The default NumPy behavior is to create arrays in either 64-bit signed
+integers or double precision floating point numbers, ``int64`` and
+``float``, respectively. If you expect your arrays to be a certain type,
+then you need to specify the ``dtype`` while you create the array. 
+
+2) Intrinsic NumPy array creation objects
+=========================================
+..
+  40 functions seems like a small number, but the routies.array-creation
+  has ~47. I'm sure there are more. 
+
+NumPy has over 40 built-in functions for creating arrays as laid
+out in the :ref:`Array creation routines <routines.array-creation>`.
+There are three categories of NumPy array creation objects:
+1) 1D arrays
+2) 2D arrays
+3) ND arrays
+
+1 - 1D array creation objects
+-----------------------------
+
+The 1D array creation objects e.g. :func:`numpy.linspace` and
+:func:`numpy.arange` generally  need at least two inputs ``start`` and
+``stop``. Where ``start`` is the first value in the array and ``stop``
+specifies the value of the last element in the array. 
+
+:func:`numpy.arange` creates arrays with regularly incrementing values.
+Check the docstring for complete information and examples. A few
+examples are shown: ::
+
+ >>> arange(10)
  array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
- >>> np.arange(2, 10, dtype=float)
+ >>> arange(2, 10, dtype=float)
  array([ 2., 3., 4., 5., 6., 7., 8., 9.])
- >>> np.arange(2, 3, 0.1)
+ >>> arange(2, 3, 0.1)
  array([ 2. , 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9])
 
-Note that there are some subtleties regarding the last usage that the user
-should be aware of that are described in the arange docstring.
+Note: there are some subtleties regarding ``dtype``. In the second
```

Comment:
Maybe here we could use a `.. note::` directive? Otherwise, the text could flow better without the colon, I think.